src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -310,16 +322,45 @@
     <MakeDir Directories="$(ProjectSpecificConfDir)" />
   </Target>
 
+    <Target Name="InvokeSonarWriteProjectDataNonRazorCompilation"
+          DependsOnTargets="SonarCategoriseProject;SonarWriteFilesToAnalyze;SonarCreateProjectSpecificDirs"
+          AfterTargets="Build;"
+          Condition=" $(SonarQubeTempPath) != '' AND '@(RazorCompile)'=='' ">
+      <CallTarget Targets="SonarWriteProjectData" />
+    </Target>
+
+    <Target Name="InvokeSonarWriteProjectDataRazorCompilation"
```

Comment:
This should also have `AfterTargets="CoreCompile"`, in theory, nothing prevents MsBuild from running this before `CoreCompile` now

Reply:
If I understand it correctly as `InvokeSonarWriteProjectDataRazorCompilation `is an AfterTarget for `SonarFinishRazorCodeAnalysis` MsBuild will not run this before `CoreCompile`. Yet to be explicit I will add the afterTarget

Reply:
`SonarFinishRazorCodeAnalysis` has `AfterTargets="RazorCoreCompile;InvokeSonarWriteProjectDataRazorCompilation"`.
So it still could be (without the last change)
```
InvokeSonarWriteProjectDataRazorCompilation
CoreCompile
RazorCoreCompile
SonarFinishRazorCodeAnalysis
```